docs/platform/how-to/timeseries-events.md

```diff
@@ -0,0 +1,103 @@
+                .add_value("PressureState", "ON") \
+                .publish()
+    else:
+        if pressure <= 0 :
+            print('State OFF')
+            triggered = False
+            stream_producer.events \
+                .add_timestamp(datetime.datetime.utcnow()) \
+                .add_value("PressureState", "OFF") \
+                .publish()
+```
+
+Here the time series data handler simply detects when the time series values exceeds the trigger thresholds, and in each case publishes an event. This event can be used to trigger processing further along in the pipeline, or the two events can be used to delimit a processing window.
+
+!!! note
+
+    In this simple example, state is not preserved across service restarts.
+
+## Handling events
+
+The above events could be handled in another service in the pipeline with the following example code:
+
+``` python
+def on_event_data_received_handler(stream_consumer: qx.StreamConsumer, data: qx.EventData):
+    if data.value == "ON":
+        print ("Process ON event")
+        ...
+    if data.value == "OFF":
+        print ("Process OFF event")
+        ...
+    ...
+```
+
+For example, you could use these events to create a time window within which you carry out processing.
+
+## Processing within the time window
+
+If you want to carry out some processing in the same service as generates the events, that can also be done. For example, to calculate the average pressure within the events:
+
+``` python
+...
+topic_consumer = client.get_topic_consumer(os.environ["pressure_values"], consumer_group = "empty-transformation")
+topic_producer = client.get_topic_producer(os.environ["pressure_events"])
+topic_averages = client.get_topic_producer(os.environ["pressure_averages"])
+
+def on_dataframe_received_handler(stream_consumer: qx.StreamConsumer, df: pd.DataFrame):
+    global triggered, average, count, total
+    stream_producer = topic_producer.get_or_create_stream(stream_id = stream_consumer.stream_id)
+    stream_averages = topic_averages.get_or_create_stream("pressure_averages")
+    pressure = df['Pressure'][0]
+    if not triggered:
+        if  pressure > 0:
+            print('State ON')
+            triggered = True
+            count = 0
+            average = 0
+            total = 0
+    else:
+        count = count + 1
+        total = (total + pressure)
+        average = total / count
+        if pressure <= 0 :
+            print('State OFF')
+            triggered = False
+            print('average : --> {:.2f}'.format(average))
+            stream_averages.timeseries.buffer \
+                .add_timestamp(datetime.datetime.utcnow()) \
+                .add_value("PressureAverage", float(average)) \
+                .publish()
+```
+
+In the above code the event generating code has been removed for simplicity. Note that the average pressure for the event window is published to an output stream, so these values can be used by other services in the pipeline.
```

Comment:
Might need a comma or something? not sure tbh

"In the above code, the event generating code has been removed"??

Reply:
Yes, I agree!